f8a_worker/workers/git_stats.py

```diff
@@ -0,0 +1,159 @@
+import shutil
+
+import numpy as np
+from tempfile import mkdtemp
+from time import time
+from datetime import timedelta
+from sklearn.linear_model import LinearRegression
+
+from f8a_worker.process import Git
+from f8a_worker.base import BaseTask
+
+
+class GitStats(BaseTask):
+    """Collect and compute Git statistics."""
+
+    _SECONDS_PER_DAY = 60 * 60 * 24
+    _DAYS_PER_MONTH = 30
+    _DAYS_PER_YEAR = 365
+
+    @staticmethod
+    def _get_log(url):
+        """Clone Git repo and get its log.
+  
+        :param url: url to the git repo
+        """
+        dir = mkdtemp()
+
+        try:
+            git = Git.clone(url, dir)
+            # nice notebook to check at:
+            #   http://nbviewer.jupyter.org/github/tarmstrong/code-analysis/blob/master/IPythonReviewTime.ipynb
+            log = git.log()
+        finally:
+            shutil.rmtree(dir, ignore_errors=True)
```

Comment:
You can use [tempdir](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/utils.py#L141):

```python
with tempdir as dir:
    git = Git.clone(url, dir)
    log = git.log()
```